### PR TITLE
fix arxiv html download redirect

### PR DESCRIPTION
**Pull request by Siedlerchr**
Fixes https://github.com/JabRef/jabref/issues/4913

We encountered a 301 redirect and thus content type was html

### Mandatory checks

- [ ] Change in `CHANGELOG.md` described in a way that is understandable for the average user (if applicable)
- [ ] Tests created for changes (if applicable)
- [ ] Manually tested changed features in running JabRef (always required)
- [ ] Screenshots added in PR description (for UI changes)
- [ ] [Checked developer's documentation](https://devdocs.jabref.org/): Is the information available and up to date? If not, I outlined it in this pull request.
- [ ] [Checked documentation](https://docs.jabref.org/): Is the information available and up to date? If not, I created an issue at <https://github.com/JabRef/user-documentation/issues> or, even better, I submitted a pull request to the documentation repository.

### CODE REVIEW

**Comment by koppor** [comment]
Tested with `1207.0408v1`. Directly downloads the PDF when fetching the entry. Nice.

File: src/main/java/org/jabref/logic/net/URLDownload.java [hunk 1]
```diff
@@ -103,15 +110,26 @@ public URL getSource() {
...
+            do {
+                HttpResponse<String> response = Unirest.head(urlToCheck).asString();
+                // Check if we have redirects, e.g. arxiv will give otherwise content type html for the original url
+                // We need to do it "manually", because ".followRedirects(true)" only works for GET not for HEAD
+                locationHeader = response.getHeaders().getFirst("location");
+                if (StringUtil.isNullOrEmpty(locationHeader)) {
+                    urlToCheck = locationHeader;
+                }
+                // while loop, because there could be multiple redirects
+            } while (StringUtil.isNullOrEmpty(locationHeader));
```
Comment (Siedlerchr) [inline src/main/java/org/jabref/logic/net/URLDownload.java:129]: we need to set a limit to max redirects ...

**Comment by koppor** [comment]
I am kind of fighting with Wiremock...  -- currently getting 404 at org.jabref.gui.linkedfile.DownloadLinkedFileActionTest#removesHtmlFileLink

**Comment by Siedlerchr** [comment]
Isidore fetcher returns two different entries... and IEEE gives bad gateway

File: src/test/java/org/jabref/gui/linkedfile/DownloadLinkedFileActionTest.java [hunk 2]
```diff
@@ -144,10 +161,20 @@ void doesntReplaceSourceURL(boolean keepHtml) throws Exception {
...
+                .willReturn(aResponse()
+                        .withStatus(200)
+                        .withHeader("Content-Type", "text/html; charset=utf-8")
+                        .withBody("<html><body><h1>Hi</h1></body></html>")));
+
+        stubFor(head(urlEqualTo("/html"))
+                .willReturn(aResponse()
+                        .withStatus(200)
+                        .withHeader("Content-Type", "text/html; charset=utf-8")
+                        .withBody("<html><body><h1>Hi</h1></body></html>")));
```
[the remark this answers is not in the document]
Comment (Siedlerchr) [reply src/test/java/org/jabref/gui/linkedfile/DownloadLinkedFileActionTest.java:175]: no, it's needed for HEAD!

[the remark this answers is not in the document]
Comment (Siedlerchr) [reply src/test/java/org/jabref/gui/linkedfile/DownloadLinkedFileActionTest.java:175]: Nope, I needed to add this explicitly for head

File: src/test/java/org/jabref/gui/linkedfile/DownloadLinkedFileActionTest.java [hunk 3]
```diff
@@ -171,10 +198,20 @@ void keepsHtmlEntry(@TempDir Path tempFolder) throws Exception {
...
```
[the remark this answers is not in the document]
Comment (Siedlerchr) [reply src/test/java/org/jabref/gui/linkedfile/DownloadLinkedFileActionTest.java:212]: Nope, I needed to add this explicitly for head

[the remark this answers is not in the document]
Comment (Siedlerchr) [reply src/test/java/org/jabref/gui/linkedfile/DownloadLinkedFileActionTest.java:212]: or do you mean the body?

**Review by koppor — APPROVED** [review]

**Pull request merged**